### PR TITLE
Anonymous key be an enum of ed25519 and secp256k1

### PR DESCRIPTION
**Pull request by findora-sun**
* **The major changes of this PR**

* **The major impacts of this PR**
  - [ ] Impact WASM?
  - [ ] Impact mainnet data compatibility?

* **Extra documentations**

### CODE REVIEW

File: api/src/anon_xfr/address_folding_ed25519.rs [hunk 1]
```diff
@@ -140,6 +139,9 @@ pub fn prove_address_folding_in_cs_secp256k1(
     secret_key_scalars_vars: &[VarIndex; 2],
     witness: &AXfrAddressFoldingWitnessEd25519,
 ) -> Result<()> {
+    //let (sk, pk) = witness.keypair.to_zorro()?; // TODO change to this
```
Comment (findora-crypto) [inline api/src/anon_xfr/address_folding_ed25519.rs:142]: TODO

File: api/src/keys.rs [hunk 2]
```diff
@@ -545,151 +761,135 @@ impl XfrKeyPair {
...
+#[derive(Clone, Debug, Eq, PartialEq)]
+/// The signature for confidential transfer.
+pub enum Signature {
+    /// Ed25519 Signature
+    Ed25519(Ed25519Signature),
+    /// Secp256k1 Signature
+    Secp256k1(Secp256k1Signature, RecoveryId),
+    /// Secp256k1 Signature with recovery.
+    /// params is r, s, v
+    Address(Secp256k1Signature, RecoveryId),
```
Comment (findora-crypto) [inline api/src/keys.rs:802]: To do: think more about what the best treatment for `Address` would be: (1) distinguish it from Ed25519 as there could be multiple formats of addresses as well---the name needs to be changed (and we may need to attach an explicit index for it)---and we need to support more and (2) see if we can reduce the duplication of code related to Address by having a function that converts it into a simple structure.

Comment (findora-sun) [reply api/src/keys.rs:802]: y, we can change `Address` to `EthAddress` explain that this is an ETH-compatible address format. and we can delete the `SecretKey::Address` and `Signature::Address` enum type, and keep the `PublicKey::EthAddress` for use eth-address to verify.

Comment (findora-crypto) [reply api/src/keys.rs:802]: we will do it in a separate PR

Comment (findora-sun) [reply api/src/keys.rs]: Just fixed this.

File: api/src/anon_xfr/address_folding_ed25519.rs (hunk 1 above)
Comment (findora-sun) [reply api/src/anon_xfr/address_folding_ed25519.rs:142]: OK

File: api/src/keys.rs (hunk 2 above)
Comment (findora-sun) [reply api/src/keys.rs:802]: OK, I revert the commits and will open new PR to fix it.

**Review by findora-crypto — APPROVED** [review]

**Pull request merged**